addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Device.java

```diff
@@ -0,0 +1,352 @@
+            logger.info("readable;writeable;recordable;virtual;type;service;value;allowed;min;max");
+            printAllServices(serviceTreeMap);
+            logger.info("##################################################################");
+        }
+
+    }
+
+    /**
+     * This function outputs a ";" separated list of all on the device available services with its capabilities
+     *
+     * @param actTreeMap
+     */
+
+    public void printAllServices(HashMap<String, KM200CommObject> actTreeMap) {
+        if (actTreeMap != null) {
+            for (KM200CommObject object : actTreeMap.values()) {
+                if (object != null) {
+                    String val = "", type, valPara = "";
+                    logger.debug("List type: {} service: {}", object.getServiceType(), object.getFullServiceName());
+                    type = object.getServiceType();
+                    if (type == null) {
+                        type = new String();
+                    }
+                    if (type.equals("stringValue") || type.equals("floatValue")) {
+                        val = object.getValue().toString();
+                        if (object.getValueParameter() != null) {
+                            if (type.equals("stringValue")) {
+                                @SuppressWarnings("unchecked")
+                                List<String> valParas = (List<String>) object.getValueParameter();
+                                for (int i = 0; i < valParas.size(); i++) {
+                                    if (i > 0) {
+                                        valPara += "|";
+                                    }
+                                    valPara += valParas.get(i);
+                                }
+                                valPara += ";;";
+                            }
+                            if (type.equals("floatValue")) {
+                                @SuppressWarnings("unchecked")
+                                List<Float> valParas = (List<Float>) object.getValueParameter();
+                                valPara += ";";
+                                if (valParas.size() == 2) {
+                                    valPara += valParas.get(0);
+                                    valPara += ";";
+                                    valPara += valParas.get(1);
+                                } else {
+                                    logger.debug("Value parameter for float != 2, this shouldn't happen");
+                                    valPara += ";";
+                                }
+                            }
+                        } else {
+                            valPara += ";;";
+                        }
+                    } else {
+                        val = "";
+                        valPara = ";";
+                    }
+                    logger.info("{};{};{};{};{};{};{};{}", object.getReadable().toString(),
```

Comment:
I would like to let it inside. The problem is that every heating system is a bit different. I'm collecting it from users to see where are the differences. It's fast and userfriendly without debug.